_posts/2020-06-12-coding-with-kids.md

```diff
@@ -0,0 +1,195 @@
+
+In overall Python is a really good language for learning programming. There is plenty of sources. It is supported in many situations, and it is simple to get started. It comes with much less boilerplate code than for example Java or C# would come, which helps kids to focus into the actual thing and not to be distracted about "important but not necessary" features of many other languages.  
+
+#### Using book to learn Python
+
+I considered how to learn programming when you already knew Scratch, PowerShell and Turtle Roy. I decided that I could go into more comprehensive direction and lend a book from library. As always with programming books, there were lots of words, and we were not interested on every of them. We cherrypicked example projects from there that were written in english and translated them with my kid to finnish. One of the examples was using `from turtle import *` to make a drawing robot that knew how to build a house, so you could command Python from commandline to provide exactly same house over and over again. That was a fun process, but the end result is rather boring. Instead, I will present you a piece software that was faster to implement and provided much more fun on many occasions: generating funny phrases. The idea is really simple. Make a few arrays of words like subjects, deeds and objects and mix them randomly to provide unexpected phrases. Use your family names in subjects to make it even funnier. Here is a hastily from finnish-to-english translated version of the implementation.
+
+```python
+subjects = ["Dad","Mother","Kid1","Kid2"]
+deeds = ["eats","kicks","burns","lifts"]
+objects = ["lion","bicycle","airplane","ice cream"]
+
+from random import randint
+def pickword(words):
+    wordcount = len(words)
+    pickedword = randint(0, wordcount-1)
+    return words[pickedword]
+
+print(pickword(subjects),pickword(deeds),pickword(objects),end='.\n')
+```
+
+This will lead into phrases like `Dad eats lion` and `Mother kicks airplane` or the sinful deed `Dad burns ice cream`. 
+
+#### RPG with Python
+
+Quickly our learning how to implement software with Python turned into how to control characters in role playing game with Python. We have been using [Code Combat](https://codecombat.com/) for that. Beside Python, it also supports JavaScript. Kids will quite quickly get grip on the game, but there are missing translations and some new features that will require a parent to support the kid on the way. Code Combat is implemented in a way that supports problem solving, learning algorithms and learning basics of object oriented programming. It is implemented with kind of Domain Specific Language. You can only use given commands and equipment will give your character more options. For example your hero can't cleave with sword without a sword. 
```

Comment:
role-playing game

Besides Python, 

problem-solving

object-oriented

with a kind

for example, (comma)

a sword

Unfortunately, (comma)

in a few years